File: py/GraphDijkstra.py
from Node import *

class GraphDijkstra:

    def __init__(self):
        self.nb_nodes = 0
        self.nodes = {} #dic
        self.nb_edges = 0

    # def initNodes(self):
    #     self.nodes = [None for _ in range(self.nb_nodes)]

    def setNbEdges(self, nb_e):
        self.nb_edges = nb_e

    def createNode(self, ID, lat, lon):
        new_node = Node(ID, lat, lon)
        # self.nodes.append(new_node)
        self.nodes[ID] = new_node

    def getNbNodes(self):
        return len(self.nodes)

    def setNbNodes(self, nb_n):
        self.nb_nodes = nb_n

    def getNode(self, i):
        return self.nodes[i]

    def visited(self, i):
        self.nodes[i].mark(True)

    def showStats(self):
        print("- Nb nodes : ", len(self.nodes))
        print("- Nb edges : ", self.nb_edges)

    def show(self):
        print("Graph : ")
        for node in self.nodes:
            # print("Node {0}, adj : ".format(node.getID()))
            # for j in range(node.getNbAdjacents()):
            #     print(node.getAdjacentNode(j).getID(), end= ", ")
            # print()
            # print("marked ", node.getMark())
            print(nodes[node])

File: version_c++/GraphDijkstra.cpp
//
// Created by Alexandre on 06-11-20.
//

#include "GraphDijkstra.h"
#include <iostream>
#include <vector>

#define G_INT_MAXI 10000000

using namespace std;

// ========================== GraphDijkstra:: =================================

// constructor defined in hpp

// getter
int GraphDijkstra::getNodesSize() const {return _nodes.size();}
Node* GraphDijkstra::getNode(int i) const {return _nodes.at(i);}

// Other
void GraphDijkstra::createNode(int id, float lat, float lon) {
    Node* new_node = new Node(id, lat, lon);
    _nodes.push_back(new_node);
}

void GraphDijkstra::show() const {
    std::cout << "Graph : " << std::endl;
    for (int i = 0; i < _nodes.size(); i++) {
        std::cout << "Node " << _nodes.at(i)->getId() << ", adj : ";
        for (int j = 0; j < _nodes.at(i)->getNbAdjacents(); j++) {
            std::cout << _nodes.at(i)->getAdjacentNode(j)->getId() << ", ";
        }
        std::cout << "marked : " << _nodes.at(i)->getMark();
        std::cout << endl;;
    }
}
